File: objectModel/TypeScript/__test__/Cdm/CdmCollection/CdmCollectionHelperFunctions.ts
// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License. See License.txt in the project root for license information.

import { testHelper } from '../../testHelper';
import {
    CdmCorpusDefinition,
    CdmDocumentDefinition,
    CdmEntityDefinition,
    CdmFolderDefinition,
    CdmManifestDefinition,
    cdmObjectType
} from '../../../internal';

export function generateManifest(): CdmManifestDefinition {
    const cdmCorpus: CdmCorpusDefinition = testHelper.getLocalCorpus(null, 'generateManifest', undefined, false, undefined, true);

    const manifest: CdmManifestDefinition = new CdmManifestDefinition(cdmCorpus.ctx, 'manifest');
    manifest.folderPath = '/';
    manifest.namespace = 'local';

    return manifest;
}

export function createDocumentForEntity(
    cdmCorpus: CdmCorpusDefinition,
    entity: CdmEntityDefinition,
    nameSpace?: string
): CdmDocumentDefinition {
    if (!nameSpace) {
        nameSpace = 'local';
    }
    const cdmFolderDef: CdmFolderDefinition = cdmCorpus.storage.fetchRootFolder(nameSpace);
    const entityDoc: CdmDocumentDefinition = cdmCorpus.MakeObject<CdmDocumentDefinition>(cdmObjectType.documentDef, `${entity.entityName}.cdm.json`, false);
    cdmFolderDef.documents.push(entityDoc);
    entityDoc.definitions.push(entity);

    return entityDoc;
}

File: objectModel/Python/tests/cdm/cdm_collection/cdm_collection_helper_functions.py
# Copyright (c) Microsoft Corporation. All rights reserved.
# Licensed under the MIT License. See License.txt in the project root for license information.

from cdm.objectmodel import CdmCorpusDefinition, CdmManifestDefinition
from cdm.enums import CdmObjectType
from tests.common import TestHelper


def generate_manifest() -> 'CdmManifestDefinition':
    """
        Creates a manifest used for the tests.
    """
    cdmCorpus = TestHelper.get_local_corpus(None, generate_manifest.__name__, no_input_and_output_folder=True)

    manifest = CdmManifestDefinition(cdmCorpus.ctx, 'manifest')
    manifest._folder_path = '/'
    manifest._namespace = 'local'

    return manifest


def create_document_for_entity(cdm_corpus: 'CdmCorpusDefinition', entity: 'CdmEntityDefinition',
                               nameSpace: str = 'local'):
    """
        For an entity, it creates a document that will contain the entity.
    """
    cdm_folder_def = cdm_corpus.storage.fetch_root_folder(nameSpace)
    entity_doc = cdm_corpus.ctx.corpus.make_object(CdmObjectType.DOCUMENT_DEF, '{}.cdm.json'.format(entity.entity_name),
                                                   False)

    cdm_folder_def.documents.append(entity_doc)
    entity_doc.definitions.append(entity)
    return entity_doc
